File: SundayAlgorithm/src/leetcode/pb23.java
package leetcode;

class pb23 {
    static List<Integer> list = new ArrayList();
    public ListNode mergeKLists(ListNode[] lists) {
        list = new ArrayList();
        for(ListNode node : lists) {
            getValue(node);
        }
        
        Collections.sort(list);
        
        ListNode answer = new ListNode(0);
        ListNode temp = answer;
        
        for(int i=0; i<list.size(); i++) {
            temp.next = new ListNode(list.get(i));
            temp = temp.next;
        }
        return answer.next;
    }
    
    public void getValue(ListNode node) {
        while(node != null) {
            list.add(node.val);
            node = node.next;
        }
    }
}

File: SundayAlgorithm/src/leetcode/pb23.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode[]} lists
 * @return {ListNode}
 */
// Merge k Sorted Lists
var array = [];
var mergeKLists = function(lists) {
    for(var i=0; i<lists.length; i++) {
        getValue(lists[i]);
    }
        
    array.sort((a,b) => b-a);
    
    let answer = new ListNode(0);
    let temp = answer;
    
    while(array.length > 0) {
        temp.next = new ListNode(array.pop());
        temp = temp.next;
    }
    return answer.next;
};

function getValue(node) {
    while(node) {
        array.push(node.val);
        node = node.next;
    }
}